llvm/lib/Analysis/ModuleSummaryAnalysis.cpp

```diff
@@ -129,15 +131,11 @@ static bool findRefEdges(ModuleSummaryIndex &Index, const User *CurUser,
   if (I) {
     uint32_t ActualNumValueData = 0;
     uint64_t TotalCount = 0;
-    // 24 is the maximum number of values preserved for one instrumented site,
-    // defined by INSTR_PROF_DEFAULT_NUM_VAL_PER_SITE in
-    // compiler-rt/lib/profile/InstrProfilingValue.c; passing 24 as
-    // `MaxNumValueData` controls the max number of elements in the returned
-    // array. The actual number of values is gated by the number of ops in !prof
-    // metadata.
+    // MaxNumVTableAnnotations is the maximum number of vtables annotated on
+    // the instruction.
     auto ValueDataArray = getValueProfDataFromInst(
-        *I, IPVK_VTableTarget, 24 /* MaxNumValueData */, ActualNumValueData,
-        TotalCount);
+        *I, IPVK_VTableTarget, MaxNumVTableAnnotations /* MaxNumValueData */,
```

Comment:
I think you can remove the "/* MaxNumValueData */" now that this is not a constant. Or if you want to keep it use the format shown in my other comment.

Reply:
done.